Missing xdg-shell-client-protocol.h on fedora with fyne v2.8.0

### Checklist

- [x] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same problem, before opening a new one.
- [x] This issue only relates to a single bug. I will open new issues for any other problems.

### Describe the bug

When updating my project from fyne 2.7.4 to 2.8.0, it no longer compiles with the following error:
```
$ fyne build
# github.com/go-gl/glfw/v3.4/glfw
In file included from vendor/github.com/go-gl/glfw/v3.4/glfw/c_glfw_lin_wayland.go:7:
./glfw/src/wl_init.c:44:10: fatal error: xdg-shell-client-protocol.h: No such file or directory
   44 | #include "xdg-shell-client-protocol.h"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.

exit status 1
make: *** [Makefile:24: build] Fehler 1
```

### How to reproduce

1. Install dependencies from [quickstart](https://docs.fyne.io/started/quick/): `sudo dnf install golang golang-misc gcc libXcursor-devel libXrandr-devel mesa-libGL-devel libXi-devel libXinerama-devel libXxf86vm-devel libxkbcommon-devel wayland-devel`
2. Upgrade project to fyne 2.8.0
3. Compile

### Screenshots

_No response_

### Example code

```
package main

import (
	"fyne.io/fyne/v2"
	"fyne.io/fyne/v2/app"
)

func main() {
	a := app.NewWithID("testapp")
	w := a.NewWindow("Test App")
	w.Resize(fyne.NewSize(800, 600))

	w.ShowAndRun()
}
```

### Fyne version

2.8.0

### Go compiler version

go version go1.26.5-X:nodwarf5 linux/amd64

### Operating system and version

Fedora 44

### Additional Information

_No response_

Huh according to AI, the header is not provided by a package but rather instead supposed to be auto-generated:
```
wayland-scanner client-header \
  /usr/share/wayland-protocols/stable/xdg-shell/xdg-shell.xml \
  xdg-shell-client-protocol.h
```
That command does indeed generate a header file for me.
Copying it to `/usr/include` does change the error for me to the next missing header file:
```
$ fyne build
# github.com/go-gl/glfw/v3.4/glfw
In file included from vendor/github.com/go-gl/glfw/v3.4/glfw/c_glfw_lin_wayland.go:7:
./glfw/src/wl_init.c:45:10: fatal error: xdg-decoration-unstable-v1-client-protocol.h: No such file or directory
   45 | #include "xdg-decoration-unstable-v1-client-protocol.h"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.

exit status 1
make: *** [Makefile:24: build] Fehler 1
```
Though these files should either be generated by the fyne command, or not included.

--- a/internal/driver/glfw/wayland_vendor_fix.go
+++ b/internal/driver/glfw/wayland_vendor_fix.go
@@ -1,0 +1,5 @@
+//go:build required
+
+package glfw
+
+import _ "github.com/go-gl/glfw/v3.4/glfw/glfw/include"

